core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -86,7 +86,10 @@ private[spark] object TaskDescription {
     dataOut.writeInt(taskDescription.properties.size())
     taskDescription.properties.asScala.foreach { case (key, value) =>
       dataOut.writeUTF(key)
-      dataOut.writeUTF(value)
+      // SPARK-19796 -- writeUTF doesn't work for long strings, which can happen for property values
+      val bytes = value.getBytes("utf-8")
+      dataOut.writeInt(bytes.length)
+      dataOut.write(bytes)
```

Comment:
Also if you do think it makes sense to use writeObject, would you mind quickly checking the size of the serialized task in this test with your original version and with writeObject just to make sure the size is comparable?